course_access_groups/acl_backends.py

```diff
@@ -1,19 +1,72 @@
 """
 Access Control backends to implement the Course Access Groups.
 """
-import six
 
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from organizations.api import get_course_organizations
+from organizations.models import UserOrganizationMapping
+from course_access_groups.models import (
+    CourseAccessGroup,
+    GroupCourse,
+    Membership,
+)
 
-def dummy_backend(user, resource, default_has_access, options):  # pylint: disable=unused-argument
+
+def is_organization_staff(user, course):
+    """
+    Helper to check if the user is organization staff.
+
+    TODO: Handle single-site setups in which organization is not important
+    """
+    # TODO: What if a course has two orgs? data leak I guess?
+    organizations = get_course_organizations(course.id)
+    if not organizations:
+        return False
+
+    return UserOrganizationMapping.objects.filter(
+        user=user,
+        organization_id__in=[org['id'] for org in organizations],
```

Comment:
At first I wondered why this didn't just call `organizations__in=organizations`. Then I read the `edx-organization` code and saw that serialized organization data was created.  It really may be better for us to do a direct query to the `OrganizationCourse` model since the model is the source of truth and we skip over a couple less levels of abstraction we don't need, as well as avoid overhead of serializing organization data:

```
org_ids = CourseOrganization.objects.filter(course_id=str(course.id)).values_list('organization__id', flat=True)
return UserOrganizationMalling.objects.filter(user=user, organization_id__in=org_ids, is_active=True, is_amc_admin=True)
```

Reply:
Good one. Done.